langchain4j/src/main/java/dev/langchain4j/agent/tool/ToolExecutor.java

```diff
@@ -1,125 +1,105 @@
-                method.setAccessible(true);
-                String result = execute(arguments);
-                log.debug("Tool execution result: {}", result);
-                return result;
-            } catch (IllegalAccessException e2) {
-                throw new RuntimeException(e2);
-            } catch (InvocationTargetException e2) {
-                Throwable cause = e2.getCause();
-                log.error("Error while executing tool", cause);
-                return cause.getMessage();
-            }
-        } catch (InvocationTargetException e) {
-            Throwable cause = e.getCause();
-            log.error("Error while executing tool", cause);
-            return cause.getMessage();
+            method.setAccessible(true);
+            return method.invoke(object, args);
         }
     }
 
-    private String execute(Object[] arguments) throws IllegalAccessException, InvocationTargetException {
-        Object result = method.invoke(object, arguments);
-        if (method.getReturnType() == void.class) {
-            return "Success";
-        }
-        return Json.toJson(result);
-    }
+    protected final Object[] prepareArguments(Map<String, Object> argumentsMap) {
+        @NotNull final String[] parameterNames = parameterNames();
+        @NotNull final Class<?>[] parameterTypes = parameterTypes();
+        @NotNull final Object[] arguments = new Object[parameterNames.length];
 
-    private Object[] prepareArguments(Map<String, Object> argumentsMap) {
-        Parameter[] parameters = method.getParameters();
-        Object[] arguments = new Object[parameters.length];
-
-        for (int i = 0; i < parameters.length; i++) {
-            String parameterName = parameters[i].getName();
-            if (argumentsMap.containsKey(parameterName)) {
-                Object argument = argumentsMap.get(parameterName);
-                Class<?> parameterType = parameters[i].getType();
-
-                // Gson always parses numbers into the Double type. If the parameter type is not Double, a conversion attempt is made.
-                if (argument instanceof Double && !(parameterType == Double.class || parameterType == double.class)) {
-                    Double doubleValue = (Double) argument;
-
-                    if (parameterType == Float.class || parameterType == float.class) {
-                        if (doubleValue < -Float.MAX_VALUE || doubleValue > Float.MAX_VALUE) {
-                            throw new IllegalArgumentException("Double value " + doubleValue + " is out of range for the float type");
-                        }
-                        argument = doubleValue.floatValue();
-                    } else if (parameterType == BigDecimal.class) {
-                        argument = BigDecimal.valueOf(doubleValue);
-                    }
-
-                    // Allow conversion to integer types only if double value has no fractional part
-                    if (hasNoFractionalPart(doubleValue)) {
```

Comment:
> Allow conversion to integer types only if double value has no fractional part
Does it necessary? why not just earse the fraction part as (byte,short,int,long,etc).

link `ToolExecutorTest` `argument type mismatched` case